assets/frontpage/events/utils/index.js

```diff
@@ -1,10 +1,66 @@
-export const toggleEventTypeDisplay = (state, eventType) => (
-  Object.assign({}, state.eventTypes, {
-    [eventType.id]: Object.assign({}, eventType, {
-      display: !eventType.display,
-    }),
-  })
-);
+export const toggleEventTypeDisplay = (state, eventType) => {
+  let newState = state;
```

Comment:
If you want to clone an object use `Object.assign({}, state)`.
Right now the original state is also being modified.